front/migrations/20241211_parents_migrator.ts

```diff
@@ -190,7 +190,61 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
   },
   google_drive: null,
   microsoft: null,
-  github: null,
+  github: {
+    transformer: (nodeId, parents) => {
+      const repoId = parents[parents.length - 1];
+      // case where we are already good
+      if (repoId.startsWith("github-repository")) {
+        return { parents, parentId: parents[1] };
+      }
+      if (parents.length < 3) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Invalid Github parents"
+        );
+      }
+      assert(/^\d+$/.test(repoId), `Invalid repoId: ${repoId}`);
+
+      if (nodeId.startsWith("github-issue-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-issues-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-discussion-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-discussions-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-code")) {
+        // github-code-${repoId}-file-xxx, github-code-${repoId}-dir-xxx, ..., github-code-${repoId}, ${repoId}
+        return {
+          parents: [
+            ...parents,
+            ...parents.slice(1, -2).reverse(),
+            `github-code-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      throw new Error(`Unrecognized node type: ${nodeId}`);
+    },
+    cleaner: (nodeId, parents) => ({ parents, parentId: parents[1] }), // TODO(2024-12-17 aubin)
```

Comment:
i'd do the cleaner here, because it will be easier to ensure that given what was done in the transformer, the cleaner will work; wdyt?